homeassistant/components/yeelock/device.py

```diff
@@ -0,0 +1,238 @@
+"""Yeelock device."""
+
+from binascii import hexlify
+import hashlib
+import hmac
+import logging
+from time import time
+import uuid
+
+from bleak import BleakClient
+from bleak.exc import BleakError
+
+from homeassistant.components import bluetooth
+from homeassistant.const import CONF_API_KEY, CONF_MAC, CONF_MODEL, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN, LOCKER_KIND, UUID_COMMAND, UUID_NOTIFY
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class YeelockDeviceEntity:
+    """Entity class for the Yeelock devices."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, yeelock_device, hass: HomeAssistant):
+        """Init entity with the device."""
+        self.hass = hass
+        self.device: Yeelock = yeelock_device
+        self._attr_unique_id = f"{yeelock_device.mac}_{self.__class__.__name__}"
+
+    @property
+    def device_info(self):
+        """Shared device info information."""
+        return {
+            "identifiers": {(DOMAIN, self.device.mac)},
+            "connections": {(dr.CONNECTION_NETWORK_MAC, self.device.mac)},
+            "name": self.device.name,
+            "manufacturer": self.device.manufacturer,
+            "model": self.device.model,
+        }
+
+
+class Yeelock:
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices